src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,183 @@
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get('id');
+    const type = urlParams.get('dataType');
+    fetch('/getrun?id=' + id + '&dataType=' + type)
+    .then(response => response.json())
+    .then(data => pose = data)
+    .then(() => addTerrian());
+}
+
+/**
+ * Creates the 2D terrain and points a light at it.
+ * It also calls two other functions so it can wait on the fetch 
+ * request and significantly cut down rendering time with 
+ * geometry instancing.
+*/
+function addTerrian() {
+  // Creates the 2D terrain.
+  dataLength = pose.length;
+  let loader = new THREE.TextureLoader();
+  let material = new THREE.MeshLambertMaterial({ 
+    map: loader.load(
+      'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=48.129872,11.582905&zoom=18&size=500x500&key='
+      + apiKey)
+    });
+  let geometry = new THREE.PlaneGeometry(50,50);
+  const map = new THREE.Mesh(geometry, material);
+  map.position.set(0,0,0);
+  map.rotation.x = -Math.PI / 2;
+  scene.add(map);
+
+  // Add the light to the scene.
+  let light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 100, 0);
+  scene.add(light)
+  console.log(dataLength);
+  plotTrajectory();
+  
+  // Adds pose data to the scene as one Geometry instance.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation = new THREE.InstancedMesh(geometry, material, dataLength);
+  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation);
+  plotOrientation();
+}
+
+/**
+ * This uses the pose data to create a blue line representing the pose
+ * trajectory. 
+ */
+function plotTrajectory() {
+  // Removes any existing trajectory objects for repositioning.
+  scene.remove(trajectory);
+  let coordinates=[];
+  for (let increment = 0; increment < dataLength; increment++) {
+    coordinates.push(new THREE.Vector3(
+      (pose[increment].lng-11.582905) * 25000 * poseScalar.multiplier,
+      (pose[increment].alt - 6.582905)/4,
+      (pose[increment].lat-48.129872) * 25000 * -poseScalar.multiplier));// GPS points
+  }
+  const geometry = new THREE.BufferGeometry().setFromPoints(coordinates);
+  const material = new THREE.LineBasicMaterial({color: 'blue'});
+  trajectory = new THREE.Line(geometry, material);
+  scene.add(trajectory);
+  trajectory.position.x = posePosition.x;
+  trajectory.position.z = posePosition.z;
+  trajectory.rotation.y = THREE.Math.degToRad(poseRotation.degrees);
+}
+
+/**
+ * This repositions all children the orientation object. The object 
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than rather than the number of data points. (O(1) vs. O(N))
+ */
+function plotOrientation() {
+  let dummy = new THREE.Object3D;
```

Comment:
What's the purpose of `dummy`? It might need a more descriptive name and a comment.